compiler/control/OMROptions.cpp

```diff
@@ -2690,7 +2692,19 @@ OMR::Options::jitPreProcess()
 
 
 #if defined(TR_HOST_64BIT) && !defined(J9ZOS390)
-       self()->setOption(TR_EnableCodeCacheConsolidation);
+      self()->setOption(TR_EnableCodeCacheConsolidation);
+#elif defined(TR_HOST_64BIT) && defined(J9ZOS390)
+      
+      // If we enabled the 2GB bar then addresses will now be from 0x80000000 to 0x7fffffffffffffff
+      // The way zOS allocates memory is by looking for 2MB free chunks and uses that for its 
+      // code caches. When we enable the 2GB bar then the addresses will be from 0x80000000
+      // to 0x7fffffffffffffff. This means two code caches could be very far apart from one another
+      // which is something we want to avoid as it may cause performance issues. The solution will 
+      // be to enable the code cache consolidation which will allocate 256MB at startup just for 
+      // code caches. This way the code caches will be together and improve performance. 
+      // Note: If all 256MB are used up then the program will abort compiles.
```

Comment:
In this comment we should replace "2GB bar" with "RMODE(64)" because we are talking about resident mode virtual addresses.